### DISCUSSION TITLE
How to use the server API ?

### DISCUSSION THREAD

**Discussion by bekreativ** [nuxt/nuxt#17417] [Questions]
Hello,

I would like to use the "server/api" folder, but I don't find many documentation or example. Maybe I missed something...

What should be the content of the js file in the api folder to be able to get de passed params/datas, and is it possible to define if the endpoint must be used with a POST, or a GET ?

Thanks for your help.

**Comment by wobsoriano** [comment]
https://v3.nuxtjs.org/guide/features/server-routes#example

**Comment by Flanker72** [comment]
There is an official documentation: https://v3.nuxtjs.org/guide/features/server-routes
and official examples: https://v3.nuxtjs.org/examples/server/routes

And also you can refer my article with some handy examples: https://medium.com/@flanker72/nuxt3-complex-solutions-database-integration-8df941f0fb82

**Comment by bekreativ** [comment]
Thanks you, it's more clear with these documentations and examples :)

I just have a question to understand all that I want. In my API, for example data.get.ts I want to request an external endpoint to build the response.

I tried useFetch in my API file, but I get the error [nuxt] [request error] useFetch is not defined.

Is it possible to use the fetch method in my API file ?

Thanks.

**Reply by wobsoriano** [reply]
You can't use composables outside of the setup function. What you want to use is `$fetch`.

```ts
// ~/server/api/data.get.ts
export default defineEventHandler(async (event) => {
  const todos = await $fetch('https://jsonplaceholder.typicode.com/todos') // $fetch is auto-imported
  return todos
})
```

**Reply by bekreativ** [reply]
Perfect ! Thanks :)
